BibBuddy/app/src/main/java/de/bibbuddy/AuthorRetriever.java

```diff
@@ -0,0 +1,125 @@
+package de.bibbuddy;
+
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+
+
+/**
+ * The AuthorRetriever class connects to the API and returns authors for a given book.
+ *
+ * @author Luis Moßburger
+ */
+public class AuthorRetriever {
+
+  private final String autApiUrl = "https://d-nb.info/gnd/";
+  private final String autApiParam = "/about/marcxml";
+
+  private static Document loadXmlFromString(String xml) throws Exception {
+    DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+    DocumentBuilder builder = factory.newDocumentBuilder();
+    InputSource is = new InputSource(new StringReader(xml));
```

Comment:
Please name variable meaningfully like inputSource or something like that.

Reply:
Sure, please see next commit!